### PR TITLE
Print debuggee console

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

### CODE REVIEW

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/Events.java [hunk 1]
```diff
@@ -138,16 +136,30 @@ public ThreadEvent(String reason, long threadId) {
 
         public Category category;
         public String output;
-        public HashMap<String, Object> data;
```
Comment (andxu) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/Events.java:141]: using Map instead

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    /**
+     * Start two separate threads to monitor the messages from stdout and stderr streams of the target process.
+     */
+    public void start() {
+        final int BUFFERSIZE = 4096;
+
+        BufferedReader stdout = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        Thread stdoutThread = new Thread(this.name + " Stdout Handler") {
+            public void run() {
```
Comment (andxu) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java]: duplicate code with stderrThread

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/Events.java (hunk 1 above)
Comment (testforstephen) [reply org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/Events.java:141]: The code is removed.

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java (hunk 2 above)
Comment (testforstephen) [reply org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java]: fixed.

**Comment by ansyral** [comment]
:shipit:

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    return;
+                }
+                int read = reader.read(buffer, 0, BUFFERSIZE);
+                if (read == -1) {
+                    subject.onComplete();
+                    return;
+                }
+                subject.onNext(new String(buffer, 0, read));
+            } catch (IOException e) {
+                subject.onComplete();
```
Comment (akaroml) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java]: onError when exception is thrown.

**Review by akaroml — CHANGES_REQUESTED** [review]

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java (hunk 3 above)
Comment (testforstephen) [reply org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProcessConsole.java]: fixed.

**Review by akaroml — APPROVED** [review]
Thanks.

**Pull request merged**